source/MaterialXTest/MaterialXRender/RenderUtil.cpp

```diff
@@ -290,40 +290,45 @@ bool ShaderRenderTester::validate(const mx::FilePathVec& testRootPaths, const mx
             outputPath.removeExtension();
             for (const auto& element : elements)
             {
-                mx::TypedElementPtr targetElement = element;
-                mx::OutputPtr output = targetElement->asA<mx::Output>();
-                mx::ShaderRefPtr shaderRef = targetElement->asA<mx::ShaderRef>();
-                mx::NodePtr outputNode = targetElement->asA<mx::Node>();
-                mx::NodeDefPtr nodeDef = nullptr;
+                std::vector<mx::TypedElementPtr> targetElements;
+                std::vector<mx::NodeDefPtr> nodeDefs;
+
+                mx::OutputPtr output = element->asA<mx::Output>();
+                mx::ShaderRefPtr shaderRef = element->asA<mx::ShaderRef>();
+                mx::NodePtr outputNode = element->asA<mx::Node>();
+
                 if (output)
                 {
                     outputNode = output->getConnectedNode();
                     // Handle connected upstream material nodes later on.
                     if (outputNode->getType() != mx::MATERIAL_TYPE_STRING)
                     {
-                        nodeDef = outputNode->getNodeDef();
+                        nodeDefs.push_back(outputNode->getNodeDef());
+                        targetElements.push_back(output);
                     }
                 }
                 else if (shaderRef)
                 {
-                    nodeDef = shaderRef->getNodeDef();
+                    nodeDefs.push_back(outputNode->getNodeDef());
+                    targetElements.push_back(shaderRef);
                 }
 
-                // Handle material node checking. For now only check first surface shader if any
+                // Get connected shader nodes if a material node.
                 if (outputNode && outputNode->getType() == mx::MATERIAL_TYPE_STRING)
                 {
-                    std::vector<mx::NodePtr> shaderNodes = getShaderNodes(outputNode, mx::SURFACE_SHADER_TYPE_STRING);
-                    if (!shaderNodes.empty())
+                    std::vector<mx::NodePtr> shaderNodes = getShaderNodes(outputNode);
+                    for (auto node : shaderNodes)
```

Comment:
Generalized this so all shader types, including displacementshaders, are extracted as renderable elements.